### PR TITLE
feat: collection in profile

### PR DESCRIPTION
**Pull request by LinHeMa**
## Features
1. Following Collections
   - Self(without pick Button)
   - Visitor(with pick Button)
2. Collections in profile
   - Self and Visitor are the same
3. Data fetching with 
  - pick -> objective === 'Collection' -> pickCollection
  - collections -> for 集錦
4. adjusting types for both profile and collection

## Libs
- using Swiper v.11.1.14 [Docs](https://swiperjs.com/react)

## Snapshot from Figma
- Collections
image

- picked Collections mobile (375px)
image

- picked Collections desktop (>= 960px)
image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208535200012071

### CODE REVIEW

File: packages/mesh/graphql/query/member.gql
Comment (erase2004) [inline packages/mesh/graphql/query/member.gql:1]: `GetMemberProfile` 和 `GetVisitorProfile` 重複的地方可以考慮用 graphql fragment 來合併處理

File: packages/mesh/types/profile.ts
Comment (erase2004) [inline packages/mesh/types/profile.ts:1]: type definition 請視為變數，不要使用相同的名稱

Comment (erase2004) [inline packages/mesh/types/profile.ts:1]: 請維持先宣告再使用的模式，避免閱讀順序混亂

File: packages/mesh/app/profile/_components/article-card-list.tsx [hunk 1]
```diff
@@ -20,29 +21,52 @@ function ArticleCardList({
...
+  //     return []
+  //   }
+
+  //   return (items as PickList)
+  //     .filter((item) => item?.objective === PickObjective.Collection)
+  //     .map(({ collection }) => ({
+  //       ...collection!,
+  //       id: collection?.id || '',
+  //     })) as PickCollections
+  // }
```
Comment (caesarWHLee) [inline packages/mesh/app/profile/_components/article-card-list.tsx]: 註解要保留建議加上說明，不保留記得刪除

File: packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx [hunk 2]
```diff
@@ -111,11 +114,34 @@ const ProfilePage: React.FC<ProfilePageProps> = ({ isMember }) => {
...
   }
 
-  const shouldShowComment = category !== TabCategory.BOOKMARKS || !isMember
+  const shouldShowComment = category === TabCategory.PICKS
+  const emptyElement = (category: TabCategory): React.ReactNode => {
+    if (category === TabCategory.COLLECTIONS)
+      return (
+        <>
+          {/**TODO: add redirect url*/}
+          <Button size="md" color="transparent" text="立即嘗試" />
```
Comment (caesarWHLee) [inline packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx:151]: 建立新集錦的 path 為 `/collection/new` 可以先加上去了

File: packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx [hunk 3]
```diff
@@ -137,10 +163,21 @@ const ProfilePage: React.FC<ProfilePageProps> = ({ isMember }) => {
         setCategory={setCategory}
         userType={isMember ? 'member' : 'visitor'}
       />
+      {pickCollections()?.length ? (
+        <>
+          <CollectionsCarousel pickCollections={pickCollections()} />
+          <p className="list-title bg-white px-5 pt-4 text-primary-700 md:bg-primary-700-dark md:p-10 md:pb-1 md:pt-9">
+            精選文章
```
Comment (caesarWHLee) [inline packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx]: 提問：精選文章的字眼是在有精選集錦區塊的才會顯示嗎？

File: packages/mesh/graphql/query/member.gql [hunk 4]
```diff
@@ -189,7 +230,55 @@ query GetMemberProfile($customId: String, $takes: Int!) {
...
+          resized {
+            w480
+            w800
+            w1200
+            w1600
+            w2400
+            original
+          }
+        }
+        pick: picks {
```
Comment (caesarWHLee) [inline packages/mesh/graphql/query/member.gql]: 應該沿用複數的 picks 比較符合理解

File: packages/mesh/graphql/query/member.gql [hunk 5]
```diff
@@ -248,7 +337,49 @@ query GetMemberProfile($customId: String, $takes: Int!) {
...
+    where: {
+      creator: { customId: { equals: $customId }, is_active: { equals: true } }
+    }
+  ) {
+    id
+    title
+    heroImage {
+      urlOriginal
+    }
+    pick: picks {
```
Comment (caesarWHLee) [inline packages/mesh/graphql/query/member.gql]: 應該沿用複數的 picks 比較符合理解，picks 應該也會需要是 is_active=true

L198, L246, L307 都需要補上

File: packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx [hunk 6]
```diff
@@ -137,10 +163,21 @@ const ProfilePage: React.FC<ProfilePageProps> = ({ isMember }) => {
         setCategory={setCategory}
         userType={isMember ? 'member' : 'visitor'}
       />
+      {pickCollections()?.length ? (
```
Comment (erase2004) [inline packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx]: 建議把 `pickCollections()` 的結果存起來重複使用

File: packages/mesh/app/profile/_components/article-card-list.tsx (hunk 1 above)
Comment (LinHeMa) [reply packages/mesh/app/profile/_components/article-card-list.tsx]: 在[42b7d5e](https://github.com/readr-media/Sachiel/pull/899/commits/42b7d5ed484cd6c287c257f6122eca6ba7536f8a)修改

File: packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx (hunk 3 above)
Comment (LinHeMa) [reply packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx]: [b8d9fa6](https://github.com/readr-media/Sachiel/pull/899/commits/b8d9fa6c88a02e80309c5f67b412c74f8f97398a)
確認是精選文章會跟著是否有picks內容，已修改

File: packages/mesh/graphql/query/member.gql (hunk 5 above)
Comment (LinHeMa) [reply packages/mesh/graphql/query/member.gql]: [d652ab4](https://github.com/readr-media/Sachiel/pull/899/commits/d652ab41faac910eb28825002384283b8f70ccff)
在此補上，同上留言

File: packages/mesh/graphql/query/member.gql
Comment (LinHeMa) [reply packages/mesh/graphql/query/member.gql:1]: [ce72de7](https://github.com/readr-media/Sachiel/pull/899/commits/ce72de72cdf822b7320c018c1d5becb4006d6d5c)
在這裡處理

File: packages/mesh/types/profile.ts
Comment (LinHeMa) [reply packages/mesh/types/profile.ts:1]: [da71941](https://github.com/readr-media/Sachiel/pull/899/commits/da7194184349ed859426990f3bf045fcfa53faa5)

File: packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx (hunk 6 above)
Comment (LinHeMa) [reply packages/mesh/app/profile/member/[customId]/_components/profile-page.tsx]: [04573c6](https://github.com/readr-media/Sachiel/pull/899/commits/04573c63bf6ac12006b6dbf5ab95e9cc55c4ce77)

File: packages/mesh/graphql/query/member.gql [hunk 7]
```diff
@@ -1,3 +1,90 @@
+fragment MemberBasicInfo on Member {
```
Comment (erase2004) [inline packages/mesh/graphql/query/member.gql]: fragment 應該放到 `/graphql/fragments` 底下

Comment (LinHeMa) [reply packages/mesh/graphql/query/member.gql]: [d4bf4a9](https://github.com/readr-media/Sachiel/pull/899/commits/d4bf4a949d724a902b0930ed3c3968cef6ba4db4) 處理

**Pull request merged**